### DISCUSSION TITLE
20240227-finops-for-startup/

### DISCUSSION THREAD

**Discussion by giscus** [inflearn/tech-blog-comment#51] [Comments]
# 20240227-finops-for-startup/

인프랩 기술 블로그

https://tech.inflab.com/20240227-finops-for-startup/

**Comment by DonggeunYu** [comment]
도움이 되었습니다 😋

**Comment by krapie** [comment]
양질의 글을 작성해 주셔서 정말 감사합니다 🙇‍♂️

**Comment by JeonYB** [comment]
몇 편에 나누어 게시하셔도 충분할 만큼의 양과 질입니다! 
잘 읽고 배우고 갑니다 :)

**Comment by yureutaejin** [comment]
와....어메이징합니다. 단순히 아키텍처적으로 효율적이다가 아닌, 모든 면에서 일일이 모니터링하고 분석해서 최적화하셨군요...
저도 클라우드 서비스 비용 최적화 작업 제안서를 올리고 작업 준비 중인데, 해당 포스팅을 참고하여 좀 더 구체화시켜 보겠습니다. 감사합니다.
인프랩 인프라, DevOps관련 글들은 정말 좋은 글들이 많은 것 같습니다.🙏

**Comment by sungkwangkim** [comment]
너무 잘 보았습니다. 많은 영감 받아 갑니다.👍

저희도 CDN 도입을 고려하고 있는데요.

개발환경에는 CDN 적용이 안될것 같은데요.
개발환경과 라이브환경이 달라서 생기는 이슈가 있으셨는지 궁금하며, 어떻게 이슈를 극복 하셨는지 궁금합니다.

**Reply by jakeinflab** [reply]
@sungkwangkim 
안녕하세요. 문의하신 내용에 답변드립니다 :) 

아시다시피 개발환경과 라이브 환경 간에 구조가 다르면 실제 배포 환경에서의 테스트가 어려운데요.
CDN 캐시 최적화 및 테스트를 위해서는 개발환경에서의 테스트가 필수적입니다.

이에 CloudFront에 WAF(Web Application Firewall) 정책을 설정해서, 
개발환경 CDN은 사전에 인가된 IP에서만 접근 가능하도록 ip-set을 구성했습니다.

CloudFront 뒤에 있는 ALB를 public으로 공개해야 하는데요, CF를 제외한 출처에서는 public ALB에 접근하지 못하도록
[Security Group inbound rule](https://aws.amazon.com/ko/blogs/networking-and-content-delivery/limit-access-to-your-origins-using-the-aws-managed-prefix-list-for-amazon-cloudfront/)을 구성하거나, [Origin Header Rule](https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/restrict-access-to-load-balancer.html#restrict-alb-add-custom-header)을 설정하는 것이 좋습니다.

이 외에도 개발 환경과 라이브 환경 간에 최대한 동일하게 인프라 아키텍처를 구축하기 위해
IaC를 이용하는 등의 노력을 파트 차원에서 진행 중입니다.

도움 되셨으면 좋겠습니다.
감사합니다!
